internal/metrics/system.go

```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	"github.com/jakub-dzon/k4e-device-worker/internal/configuration"
+	"github.com/jakub-dzon/k4e-operator/models"
+	"reflect"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	DefaultSystemMetricsScrapingInterval = 60
+	NodeExporterMetricsEndpoint          = "http://localhost:9100/metrics"
+)
+
+var defaultSystemMetricsConfiguration = models.SystemMetricsConfiguration{Interval: DefaultSystemMetricsScrapingInterval}
+
+//go:generate mockgen -package=metrics -destination=configuration_provider_mock.go . DeviceConfigurationProvider
+type DeviceConfigurationProvider interface {
+	GetDeviceConfiguration() models.DeviceConfiguration
+}
+
+type SystemMetrics struct {
+	latestConfig atomic.Value
+	daemon       MetricsDaemon
+	config       configuration.Manager
+}
+
+func NewSystemMetrics(daemon MetricsDaemon, config DeviceConfigurationProvider) *SystemMetrics {
+	expectedConfiguration := expectedConfiguration(config.GetDeviceConfiguration())
+
+	sm := &SystemMetrics{
+		daemon: daemon,
+	}
+	daemon.AddTarget("system", []string{NodeExporterMetricsEndpoint}, time.Duration(expectedConfiguration.Interval)*time.Second, nil)
+	sm.latestConfig.Store(&expectedConfiguration)
+	return sm
+}
+
+func (sm *SystemMetrics) Update(config models.DeviceConfigurationMessage) error {
+	newConfiguration := expectedConfiguration(*config.Configuration)
+	latestConfig := sm.latestConfig.Load()
+	if latestConfig != nil {
+		oldConfiguration := latestConfig.(*models.SystemMetricsConfiguration)
+		if oldConfiguration != nil && reflect.DeepEqual(newConfiguration, *oldConfiguration) {
+			return nil
+		}
+	}
+
+	sm.daemon.AddTarget("system", []string{NodeExporterMetricsEndpoint}, time.Duration(newConfiguration.Interval)*time.Second, nil)
```

Comment:
We should add labels here, at least system one, if not can be mixed with another workload. Also, maybe the machine-id, if not multiple devices will have the same info, no?

Reply:
The target name will be added as a label and if we don't allow using `system` workload target, it will be distinguishable.

The device ID label can be added when we send the metrics to the control plane - let's not waste memory on storing same label value multiple times.